crypto/x509/x509_vfy.c

```diff
@@ -3440,6 +3444,29 @@ static int check_key_level(X509_STORE_CTX *ctx, X509 *cert)
     return EVP_PKEY_security_bits(pkey) >= minbits_table[level - 1];
 }
 
+/*
+ * Check whether the public key of ``cert`` does not use explicit params
+ * for an elliptic curve.
+ *
+ * Returns 1 on success, 0 otherwise.
+ */
+static int check_curve(X509 *cert)
+{
+#ifndef OPENSSL_NO_EC
+    EVP_PKEY *pkey = X509_get0_pubkey(cert);
+
+    /* Unsupported or malformed keys will fail other checks, ignore them */
+    if (pkey == NULL)
+        return 1;
```

Comment:
Here we could return -1 and chack for tri-state in the caller

Reply:
> Here we could return -1 and chack for tri-state in the caller

I support the suggestion to use -1 wherever an essentially Boolean function hits a condition where it cannot yield a proper result.